packages/world/src/modules/uniqueentity/UniqueEntityModule.sol

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { IModule } from "../../interfaces/IModule.sol";
+
+import { WorldContext } from "../../WorldContext.sol";
+
+import { UniqueEntity } from "./tables/UniqueEntity.sol";
+
+/**
+ * This module creates a table that stores a nonce, and `getUniqueEntity`
+ * returns an incremented nonce each time.
+ *
+ * Note: this module currently expects to be `delegatecalled` via World.installRootModule.
+ * Support for installing it via `World.installModule` depends on `World.callFrom` being implemented.
```

Comment:
The `UniqueEntity` table doesn't have to be owned by the caller of this module. Instead we can create a new namespace for the module in which the module installs the table (and thereby owns the table). Then we can create a public system to increment the value, and add a util to call the system (instead of putting the write logic in the util directly).

Reply:
yeah the system stuff you mention is why this is a draft - I've been thinking about adding a system as well